modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheOffheapManager.java

```diff
@@ -1013,6 +1023,16 @@ private Metas getOrAllocateCacheMetas() throws IgniteCheckedException {
         if (grp.mvccEnabled()) // TODO IGNITE-7384
             return super.historicalIterator(partCntrs, missing);
 
+        GridCacheDatabaseSharedManager database = (GridCacheDatabaseSharedManager)grp.shared().database();
+
+        FileWALPointer latestReservedPointer = (FileWALPointer)database.latestWalPointerReservedForPreloading();
+
+        if (latestReservedPointer == null) {
+            log.warning("Historical iterator wasn't created, because WAL isn't reserved.");
+
+            return null;
```

Comment:
Should't we throw here IgniteHistoricalIteratorException to fallback to full rebalancing ?

Reply:
Good point, I will throw the exception here.